File: experiments/adrianco/experiment-46-opus5/brazil/runs/language=rust_model=claude-opus-5_prompt=neutral/rep1/src/samples.rs
//! The worked example questions from the specification, each bound to the tool
//! call that answers it.
//!
//! `cargo run -- demo` executes the list and prints every answer; the
//! integration tests execute the same list and assert that all of them return
//! substantive answers inside the required latency budget. Keeping one list
//! means the demo can never drift from what is tested.

use serde_json::{json, Value};

/// A natural-language question and the tool call that answers it.
pub struct SampleQuestion {
    /// The question a user would ask the LLM.
    pub question: &'static str,
    /// Capability category from the specification.
    pub category: &'static str,
    pub tool: &'static str,
    pub arguments: fn() -> Value,
}

/// Sample questions covering all five specified capability areas.
pub const SAMPLE_QUESTIONS: &[SampleQuestion] = &[
    // ---- 1. match queries ---------------------------------------------------
    SampleQuestion {
        question: "Show me all Flamengo vs Fluminense matches",
        category: "Match queries",
        tool: "search_matches",
        arguments: || json!({ "team": "Flamengo", "opponent": "Fluminense", "limit": 10 }),
    },
    SampleQuestion {
        question: "What matches did Palmeiras play in 2023?",
        category: "Match queries",
        tool: "search_matches",
        arguments: || json!({ "team": "Palmeiras", "season": 2023, "limit": 15 }),
    },
    SampleQuestion {
        question: "Find all Copa do Brasil finals",
        category: "Match queries",
        tool: "search_matches",
        arguments: || json!({ "competition": "Copa do Brasil", "stage": "8", "limit": 25 }),
    },
    SampleQuestion {
        question: "Show me the Copa Libertadores finals",
        category: "Match queries",
        tool: "search_matches",
        arguments: || json!({ "competition": "Libertadores", "stage": "final", "limit": 25 }),
    },
    SampleQuestion {
        question: "When did Flamengo last play Corinthians, and what was the score?",
        category: "Match queries",
        tool: "head_to_head",
        arguments: || json!({ "team_a": "Flamengo", "team_b": "Corinthians", "recent": 3 }),
    },
    SampleQuestion {
        question:
            "Which matches were played at the Maracanã derby between Vasco and Botafogo in 2019?",
        category: "Match queries",
        tool: "search_matches",
        arguments: || json!({ "team": "Vasco", "opponent": "Botafogo", "season": 2019 }),
    },
    SampleQuestion {
        question: "What Série A matches were played in the first week of November 2019?",
        category: "Match queries",
        tool: "search_matches",
        arguments: || {
            json!({
                "competition": "Serie A",
                "date_from": "2019-11-01",
                "date_to": "2019-11-07",
                "limit": 20
            })
        },
    },
    // ---- 2. team queries ----------------------------------------------------
    SampleQuestion {
        question: "What is Corinthians' home record in 2022?",
        category: "Team queries",
        tool: "team_stats",
        arguments: || json!({ "team": "Corinthians", "season": 2022, "venue": "home" }),
    },
    SampleQuestion {
        question: "Which team scored the most goals in Série A 2023?",
        category: "Team queries",
        tool: "team_rankings",
        arguments: || json!({ "metric": "goals_for", "competition": "Serie A", "season": 2023, "limit": 5 }),
    },
    SampleQuestion {
        question: "Compare Palmeiras and Santos head-to-head",
        category: "Team queries",
        tool: "head_to_head",
        arguments: || json!({ "team_a": "Palmeiras", "team_b": "Santos", "recent": 5 }),
    },
    SampleQuestion {
        question: "What competitions has Palmeiras played in?",
        category: "Team queries",
        tool: "team_profile",
        arguments: || json!({ "team": "Palmeiras" }),
    },
    SampleQuestion {
        question: "How did Cruzeiro do in the 2003 Brasileirão?",
        category: "Team queries",
        tool: "team_stats",
        arguments: || json!({ "team": "Cruzeiro", "season": 2003, "competition": "Serie A" }),
    },
    SampleQuestion {
        question: "What is Corinthians' record in the Copa Libertadores?",
        category: "Team queries",
        tool: "team_stats",
        arguments: || json!({ "team": "Corinthians", "competition": "Libertadores" }),
    },
    SampleQuestion {
        question: "Which club does 'Atlético' refer to?",
        category: "Team queries",
        tool: "find_team",
        arguments: || json!({ "query": "Atlético" }),
    },
    // ---- 3. player queries --------------------------------------------------
    SampleQuestion {
        question: "Find all Brazilian players in the dataset",
        category: "Player queries",
        tool: "search_players",
        arguments: || json!({ "nationality": "Brazil", "limit": 10 }),
    },
    SampleQuestion {
        question: "Who are the highest-rated players at Grêmio?",
        category: "Player queries",
        tool: "club_players",
        arguments: || json!({ "club": "Grêmio", "limit": 10 }),
    },
    SampleQuestion {
        question: "Who are the highest-rated players at Flamengo?",
        category: "Player queries",
        tool: "club_players",
        arguments: || json!({ "club": "Flamengo" }),
    },
    SampleQuestion {
        question: "Show me all strikers at Brazilian clubs",
        category: "Player queries",
        tool: "search_players",
        arguments: || json!({ "position": "ST", "brazilian_clubs_only": true, "limit": 10 }),
    },
    SampleQuestion {
        question: "Who is Neymar Jr?",
        category: "Player queries",
        tool: "player_profile",
        arguments: || json!({ "name": "Neymar Jr" }),
    },
    SampleQuestion {
        question: "Which Brazilian players at Brazilian clubs are rated 80 or above?",
        category: "Player queries",
        tool: "search_players",
        arguments: || {
            json!({
                "nationality": "Brazil",
                "brazilian_clubs_only": true,
                "min_overall": 78,
                "limit": 10
            })
        },
    },
    // ---- 4. competition queries ---------------------------------------------
    SampleQuestion {
        question: "Who won the 2019 Brasileirão?",
        category: "Competition queries",
        tool: "standings",
        arguments: || json!({ "competition": "Serie A", "season": 2019, "limit": 6 }),
    },
    SampleQuestion {
        question: "Which teams were relegated in 2020?",
        category: "Competition queries",
        tool: "standings",
        arguments: || json!({ "competition": "Serie A", "season": 2020 }),
    },
    SampleQuestion {
        question: "Show the 2018 Copa Libertadores knockout stages",
        category: "Competition queries",
        tool: "search_matches",
        arguments: || json!({ "competition": "Libertadores", "season": 2018, "stage": "final", "limit": 10 }),
    },
    SampleQuestion {
        question: "What did the 2015 Série B table look like?",
        category: "Competition queries",
        tool: "standings",
        arguments: || json!({ "competition": "Serie B", "season": 2015, "limit": 8 }),
    },
    // ---- 5. statistical analysis --------------------------------------------
    SampleQuestion {
        question: "What's the average goals per match in the Brasileirão?",
        category: "Statistical analysis",
        tool: "competition_stats",
        arguments: || json!({ "competition": "Serie A" }),
    },
    SampleQuestion {
        question: "Compare the 2018 and 2019 Série A seasons",
        category: "Statistical analysis",
        tool: "competition_stats",
        arguments: || json!({ "competition": "Serie A", "seasons": [2018, 2019] }),
    },
    SampleQuestion {
        question: "Which team has the best away record in the Brasileirão?",
        category: "Statistical analysis",
        tool: "team_rankings",
        arguments: || {
            json!({
                "metric": "win_rate",
                "competition": "Serie A",
                "venue": "away",
                "min_matches": 100,
                "limit": 5
            })
        },
    },
    SampleQuestion {
        question: "Which team has the best home record in the Brasileirão?",
        category: "Statistical analysis",
        tool: "team_rankings",
        arguments: || {
            json!({
                "metric": "win_rate",
                "competition": "Serie A",
                "venue": "home",
                "min_matches": 100,
                "limit": 5
            })
        },
    },
    SampleQuestion {
        question: "Show me the biggest wins in the dataset",
        category: "Statistical analysis",
        tool: "biggest_wins",
        arguments: || json!({ "limit": 10 }),
    },
    SampleQuestion {
        question: "How does the Libertadores compare to the Brasileirão for goals per game?",
        category: "Statistical analysis",
        tool: "competition_stats",
        arguments: || json!({ "competition": "Libertadores" }),
    },
    // ---- cross-file and graph queries ---------------------------------------
    SampleQuestion {
        question:
            "Which Brazilian clubs have both squad and match data, and how do their squads rate?",
        category: "Cross-file queries",
        tool: "search_players",
        arguments: || json!({ "brazilian_clubs_only": true, "limit": 10 }),
    },
    SampleQuestion {
        question: "What is connected to Santos in the knowledge graph?",
        category: "Cross-file queries",
        tool: "graph_neighbors",
        arguments: || json!({ "node_type": "team", "name": "Santos", "limit": 3 }),
    },
    SampleQuestion {
        question: "What data is loaded and what are its limits?",
        category: "Cross-file queries",
        tool: "dataset_overview",
        arguments: || json!({}),
    },
];

File: experiments/adrianco/experiment-55-terra-vs-opus5-effort/brazil/runs/agent=claude-code_effort=high_language=go_model=claude-opus-5_prompt=neutral/rep1/internal/mcpsrv/samples.go
// samples.go is the catalogue of natural-language questions this server is
// designed to answer, each paired with the tool call that answers it. It backs
// both the `-demo` mode of the binary and the acceptance test that checks every
// question still produces an answer.
package mcpsrv

// SampleQuestion is one natural-language question and the tool call that
// answers it.
type SampleQuestion struct {
	Question    string
	Tool        string
	Args        map[string]any
	ExpectError bool // the honest answer is "that is not in the data"
}

// SampleQuestions returns the demonstration set. It covers all five capability
// categories in the specification: match, team, player, competition and
// statistical queries, plus the cross-dataset player/match join.
func SampleQuestions() []SampleQuestion {
	return []SampleQuestion{
		// --- Match queries --------------------------------------------------
		{
			Question: "Show me all Flamengo vs Fluminense matches",
			Tool:     "search_matches",
			Args:     map[string]any{"team": "Flamengo", "opponent": "Fluminense", "limit": 10},
		},
		{
			Question: "What matches did Palmeiras play in 2023?",
			Tool:     "search_matches",
			Args:     map[string]any{"team": "Palmeiras", "season": 2023, "limit": 10},
		},
		{
			Question: "Find all Copa do Brasil finals",
			Tool:     "search_matches",
			Args:     map[string]any{"competition": "copa-do-brasil", "stage": "final", "limit": 60, "sort": "date_asc"},
		},
		{
			Question: "When did Flamengo last play Corinthians, and what was the score?",
			Tool:     "search_matches",
			Args:     map[string]any{"team": "Flamengo", "opponent": "Corinthians", "limit": 1},
		},
		{
			Question: "Which matches did Grêmio play at home in the 2019 Brasileirão?",
			Tool:     "search_matches",
			Args:     map[string]any{"team": "Gremio", "competition": "serie-a", "season": 2019, "venue": "home", "limit": 20},
		},
		{
			Question: "Show me the biggest wins in the dataset",
			Tool:     "search_matches",
			Args:     map[string]any{"sort": "goal_diff", "limit": 10},
		},

		// --- Team queries ---------------------------------------------------
		{
			Question: "What is Corinthians' home record in the 2022 Brasileirão?",
			Tool:     "team_stats",
			Args:     map[string]any{"team": "Corinthians", "competition": "serie-a", "season": 2022, "venue": "home"},
		},
		{
			Question: "What competitions has Palmeiras played in?",
			Tool:     "team_stats",
			Args:     map[string]any{"team": "Palmeiras"},
		},
		{
			Question: "Compare Palmeiras and Santos head-to-head",
			Tool:     "head_to_head",
			Args:     map[string]any{"team_a": "Palmeiras", "team_b": "Santos", "limit": 10},
		},
		{
			Question: "How do Grêmio and Internacional compare in the Gre-Nal?",
			Tool:     "head_to_head",
			Args:     map[string]any{"team_a": "Gremio", "team_b": "Internacional", "limit": 10},
		},
		{
			Question: "Which clubs are called Atlético?",
			Tool:     "search_teams",
			Args:     map[string]any{"query": "Atletico", "limit": 10},
		},

		// --- Competition queries --------------------------------------------
		{
			Question: "Who won the 2019 Brasileirão?",
			Tool:     "standings",
			Args:     map[string]any{"competition": "serie-a", "season": 2019},
		},
		{
			Question: "Which teams were relegated in 2020?",
			Tool:     "standings",
			Args:     map[string]any{"competition": "serie-a", "season": 2020},
		},
		{
			Question: "Show the 2018 Copa Libertadores bracket",
			Tool:     "competition_bracket",
			Args:     map[string]any{"competition": "libertadores", "season": 2018},
		},
		{
			Question: "What competitions and seasons are available?",
			Tool:     "list_competitions",
			Args:     map[string]any{},
		},

		// --- Statistical analysis -------------------------------------------
		{
			Question: "What is the average number of goals per match in the Brasileirão?",
			Tool:     "league_stats",
			Args:     map[string]any{"competition": "serie-a"},
		},
		{
			Question: "Which team scored the most goals in Série A in 2023?",
			Tool:     "league_stats",
			Args:     map[string]any{"competition": "serie-a", "season": 2023, "top_n": 5},
		},
		{
			Question: "Which team has the best away record in the Brasileirão?",
			Tool:     "league_stats",
			Args:     map[string]any{"competition": "serie-a", "min_matches": 100, "top_n": 5},
		},
		{
			Question: "Compare the 2018 and 2019 Brasileirão seasons",
			Tool:     "compare_seasons",
			Args:     map[string]any{"competition": "serie-a", "seasons": []int{2018, 2019}},
		},
		{
			Question: "Show me all derbies in 2023",
			Tool:     "derbies",
			Args:     map[string]any{"season": 2023, "limit": 30},
		},

		// --- Player queries -------------------------------------------------
		{
			Question: "Who are the top Brazilian players in the dataset?",
			Tool:     "search_players",
			Args:     map[string]any{"nationality": "Brazil", "sort_by": "overall", "limit": 10},
		},
		{
			Question: "How many Brazilian players are at Brazilian clubs, and how good are they?",
			Tool:     "search_players",
			Args:     map[string]any{"nationality": "Brazil", "brazilian_clubs_only": true, "group_by": "club", "limit": 10},
		},
		{
			Question: "Who are the highest-rated players at Grêmio?",
			Tool:     "club_squad",
			Args:     map[string]any{"club": "Gremio", "limit": 10},
		},
		{
			Question: "Show me all forwards at Atlético Mineiro",
			Tool:     "search_players",
			Args:     map[string]any{"club": "Atletico Mineiro", "position": "forward", "limit": 10},
		},
		{
			Question: "Who is Neymar Jr?",
			Tool:     "player_profile",
			Args:     map[string]any{"name": "Neymar"},
		},
		{
			Question: "Which players play for Flamengo?",
			Tool:     "club_squad",
			Args:     map[string]any{"club": "Flamengo"},
		},
		{
			Question:    "Who is Gabriel Barbosa?",
			Tool:        "player_profile",
			Args:        map[string]any{"name": "Gabriel Barbosa"},
			ExpectError: true, // not present in the FIFA 19 snapshot
		},

		// --- Provenance -----------------------------------------------------
		{
			Question: "What data is this server working from?",
			Tool:     "dataset_info",
			Args:     map[string]any{},
		},
	}
}
